### PR TITLE
[17.0][MIG] partner_manual_rank: Migration to 17.0

### PR DESCRIPTION
**Pull request by tien-ld**

### CODE REVIEW

**Review by jcdrubay — APPROVED** [review]
Code review, LGTM

**Comment by DucTruongKomit** [comment]
LGTM

**Review by DucTruongKomit — APPROVED** [review]
LGTM

**Review by ioans73 — APPROVED** [review]
Functional review

**Review by desdelinux — APPROVED** [review]
LGTM @luisg123v

Ready for merge

File: partner_manual_rank/README.rst [hunk 1]
```diff
@@ -17,19 +17,20 @@ Partner Manual Rank
     :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
     :alt: License: AGPL-3
 .. |badge3| image:: https://img.shields.io/badge/github-OCA%2Fpartner--contact-lightgray.png?logo=github
-    :target: https://github.com/OCA/partner-contact/tree/16.0/partner_manual_rank
+    :target: https://github.com/OCA/partner-contact/tree/17.0/partner_manual_rank
```
Comment (luisg123v) [inline partner_manual_rank/README.rst:20]: I think this should belong to the autoformatting commit, which is missing here.

File: partner_manual_rank/__manifest__.py [hunk 2]
```diff
@@ -5,7 +5,7 @@
 {
     "name": "Partner Manual Rank",
     "summary": "Be able to manually flag partners as customer or supplier.",
-    "version": "16.0.1.1.2",
+    "version": "17.0.0.0.0",
```
Comment (luisg123v) [inline partner_manual_rank/__manifest__.py]: 17.0.**1**.0.0

File: partner_manual_rank/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[build-system]
```
Comment (luisg123v) [inline partner_manual_rank/pyproject.toml:1]: Same here and all similar cases, this should be on a separate commit containing the result of running `precommit`

File: partner_manual_rank/tests/test_res_partner.py [hunk 4]
```diff
@@ -43,6 +44,16 @@ def test_01_is_customer(self):
...
         )
+        partners.write({"customer_rank": 1})
+        self.assertRecordValues(
+            partners,
+            [
+                {"is_customer": True, "customer_rank": 1},
+                {"is_customer": True, "customer_rank": 1},
+            ],
+        )
+        with self.assertRaises(UserError):
```
Comment (luisg123v) [inline partner_manual_rank/tests/test_res_partner.py]: Use

`self.assertRaisesRegex(UserError, error_msg):`

To ensure the raised error is the expected one.

Which one are you expecting here?

Same below.

**Review by luisg123v — CHANGES_REQUESTED** [review]

File: partner_manual_rank/__manifest__.py (hunk 2 above)
Comment (tien-ld) [reply partner_manual_rank/__manifest__.py]: I updated the version

File: partner_manual_rank/README.rst (hunk 1 above)
Comment (tien-ld) [reply partner_manual_rank/README.rst:20]: I split the commit into 2

File: partner_manual_rank/tests/test_res_partner.py (hunk 4 above)
Comment (tien-ld) [reply partner_manual_rank/tests/test_res_partner.py]: Hi, I added this to cover these function

image

I just updated the code to use `assertRaisesRegex`

**Comment by tien-ld** [comment]
Hi @luisg123v I updated the commit & code, could you please help to review it again, thank you

File: partner_manual_rank/README.rst [hunk 5]
```diff
@@ -63,6 +63,7 @@ Contributors
 -  Francesco Apruzzese <cescoap@gmail.com>
 -  Francisco Luna <fluna@vauxoo.com>
 -  Luis González <lgonzalez@vauxoo.com>
+-  Le Dinh Tien <tien-ld@komit-consulting.com>
```
Comment (luisg123v) [inline partner_manual_rank/README.rst]: Could you add yourself only in the file `CONTRIBUTORS.md`, please?

The rest (this README and the index files) will be taken care of by the bot.

Comment (tien-ld) [reply partner_manual_rank/README.rst]: Hi @luisg123v, during the migration progress, I only added myself in `CONTRIBUTORS.md`
the README and other files were updated by pre-commit

So can we include it in this commit? or just ignore these changes and update `CONTRIBUTORS.md` only?

Comment (luisg123v) [reply partner_manual_rank/README.rst]: AFAIK we should modify only `CONTRIBUTORS.md`, and the rest will be updated automatically after migration. This so that those changes are not included as part of the migration commit.

Could you try that, please?

Comment (tien-ld) [reply partner_manual_rank/README.rst]: yes of course, I updated the commit @luisg123v

**Review by luisg123v — APPROVED** [review]
LGTM :+1: 

/ocabot merge nobump
